subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junit/junit4/AbstractJUnit4TestListenerIntegrationTest.groovy

```diff
@@ -72,4 +72,48 @@ abstract class AbstractJUnit4TestListenerIntegrationTest extends AbstractJUnitTe
         assert containsLine(result.getOutput(), "START [Test testError(SomeTest)] [testError]")
         assert containsLine(result.getOutput(), "FINISH [Test testError(SomeTest)] [testError] [java.lang.RuntimeException: message]")
     }
+
+    def "can listen for test events using closures"() {
```

Comment:
This now works in master, but it would fail with the CC-enabled executer in 8.3 (I applied this commit to a local copy of the release branch, and got this:

```
Condition not satisfied:

containsLine(result.getOutput(), "START [Test class SomeTest] [SomeTest]")
|            |      |
false        |      > Task :processTestResources NO-SOURCE
             |      > Task :processResources NO-SOURCE
             |      > Task :compileJava NO-SOURCE
             |      > Task :classes UP-TO-DATE
             |      > Task :compileTestJava
             |      > Task :testClasses
             |       
             |      > Task :test
             |       
             |      SomeTest > testError FAILED
             |          java.lang.RuntimeException at SomeTest.java:17
             |       
             |      SomeTest > testFail FAILED
             |          java.lang.AssertionError at SomeTest.java:15
             |      There were failing tests. See the report at: file:///Users/rafael/sources/gradle/trees/release/subprojects/testing-jvm/build/tmp/te%C5%9Dt%20files/JUnit4TestL.Test/t9b6w/build/reports/tests/test/index.html
             |       
             |      BUILD SUCCESSFUL in 823ms
             |      2 actionable tasks: 2 executed
             |      Configuration cache entry stored.
             <org.gradle.integtests.fixtures.executer.OutputScrapingExecutionResult@204b0f07 output=[> Task :processTestResources NO-SOURCE, > Task :processResources NO-SOURCE, > Task :compileJava NO-SOURCE, > Task :classes UP-TO-DATE, > Task :compileTestJava, > Task :testClasses, , > Task :test, , SomeTest > testError FAILED,     java.lang.RuntimeException at SomeTest.java:17, , SomeTest > testFail FAILED,     java.lang.AssertionError at SomeTest.java:15, There were failing tests. See the report at: file:///Users/rafael/sources/gradle/trees/release/subprojects/testing-jvm/build/tmp/te%C5%9Dt%20files/JUnit4TestL.Test/t9b6w/build/reports/tests/test/index.html, , BUILD SUCCESSFUL in 823ms, 2 actionable tasks: 2 executed, Configuration cache entry stored., ] error=[, 3 tests completed, 2 failed, ] includeBuildSrc=true mainContent=[> Task :processTestResources NO-SOURCE, > Task :processResources NO-SOURCE, > Task :compileJava NO-SOURCE, > Task :classes UP-TO-DATE, > Task :compileTestJava, > Task :testClasses, , > Task :test, , SomeTest > testError FAILED,     java.lang.RuntimeException at SomeTest.java:17, , SomeTest > testFail FAILED,     java.lang.AssertionError at SomeTest.java:15, There were failing tests. See the report at: file:///Users/rafael/sources/gradle/trees/release/subprojects/testing-jvm/build/tmp/te%C5%9Dt%20files/JUnit4TestL.Test/t9b6w/build/reports/tests/test/index.html, ] postBuild=[2 actionable tasks: 2 executed, Configuration cache entry stored., ] errorContent=[, 3 tests completed, 2 failed, ] groupedOutputFixture=null tasks=null>

	at org.gradle.testing.junit.junit4.AbstractJUnit4TestListenerIntegrationTest.can listen for test events using closures(AbstractJUnit4TestListenerIntegrationTest.groovy:110)
```

Reply:
@tresat may still have some context on this from when he reviewed  https://github.com/gradle/gradle/pull/25983

Reply:
Does this mean the closure-based listeners aren't running at all?  Are they broken?

Reply:
To be clear, we are talking about configuration cache behavior.